ironfish-rust/src/serializing.rs

```diff
@@ -75,57 +73,76 @@ pub(crate) mod aead {
     ///
     /// This is just a facade around the ChaCha20Poly1305 struct. It ignores
     /// nonce and aad and automatically stores the mac tag.
-    pub(crate) fn encrypt(key: &[u8], plaintext: &[u8], encrypted_output: &mut [u8]) {
-        assert_eq!(encrypted_output.len(), plaintext.len() + MAC_SIZE);
-        let mut encryptor = ChaCha20Poly1305::new(key, &[0; 8], &[0; 8]);
-        let mut tag = [0; MAC_SIZE];
-        encryptor.encrypt(
-            plaintext,
-            &mut encrypted_output[..plaintext.len()],
-            &mut tag,
-        );
-        encrypted_output[plaintext.len()..].clone_from_slice(&tag);
+    /// This is just a facade around the ChaCha20Poly1305 struct. The nonce and
+    /// associated data are zeroed.
+    pub(crate) fn encrypt<const SIZE: usize>(
+        key: &[u8; 32],
+        plaintext: &[u8],
+    ) -> Result<[u8; SIZE], IronfishError> {
+        let mut encrypted_output = [0u8; SIZE];
+        encrypted_output[..plaintext.len()].copy_from_slice(plaintext);
+
+        let encryptor = ChaCha20Poly1305::new(Key::from_slice(key));
+
+        let tag = encryptor
+            .encrypt_in_place_detached(
+                &Nonce::default(),
+                &[],
+                &mut encrypted_output[..plaintext.len()],
+            )
+            .map_err(|_| IronfishError::InvalidSigningKey)?;
+        encrypted_output[plaintext.len()..].copy_from_slice(&tag);
+
+        Ok(encrypted_output)
     }
 
     /// Decrypt the encrypted text using the given key and ciphertext, also checking
     /// that the mac tag is correct.
     ///
     /// Returns Ok(()) if the mac matches the decrypted text, Err(()) if not
```

Comment:
```suggestion
```

Probably just dont need this line anymore, I think the type definition is clear enough